src/hookproxy/proxydll.cpp

```diff
@@ -0,0 +1,221 @@
+}
+__declspec(naked) void FakeAVIStreamRelease()
+{
+    _asm { jmp[OriginalAVIStreamRelease] }
+}
+
+/**
+ * Get the path for the exe we were loaded by.
+ */
+wchar_t *Get_Path()
+{
+    static wchar_t buffer[MAX_PATH];
+    DWORD size = GetModuleFileNameW(nullptr, buffer, ARRAY_SIZE(buffer));
+
+    if (size == 0 || size >= ARRAY_SIZE(buffer)) {
+        MessageBoxA(nullptr, "Failed to get path to target executable for code injection", "Proxy", MB_ICONERROR);
+        ExitProcess(0);
+    }
+
+    return buffer;
+}
+
+/**
+ * Get the hash for the exe we were loaded by.
+ */
+const char* Get_Hash()
+{
+    SHAEngine sha;
+    char buffer[1024];
+
+    std::ifstream in(Get_Path(), std::ios_base::in | std::ios_base::binary);
+
+    // Get the hash of the binary.
+    do {
+        in.read(buffer, sizeof(buffer));
+        sha.Hash(buffer, int(in.gcount()));
+    } while (in.gcount() == sizeof(buffer));
+
+    return sha.Print_Result();
+}
+
+/**
+ * Get the payload dll name we expect for the binary based on its hash.
+ */
+const char* Get_Payload_Name()
+{
+    struct ExePayloads
+    {
+        const char *exe_hash;
+        const char *payload_name;
+    };
+    
+    static const ExePayloads Payloads[] = {
+        { "5b805b4ecba79581fe8ec454b5acf7b7e13d4dd5", "thyme.dll" },
+    };
+
+    const char *target_hash = Get_Hash();
+
+    for (unsigned i = 0; i < ARRAY_SIZE(Payloads); ++i) {
+        if (strcmp(Payloads[i].exe_hash, target_hash) == 0) {
+            return Payloads[i].payload_name;
+        }
+    }
+
+    return nullptr;
+}
+
+/**
+ * Loads the dll we are proxying for and the functions the target application imports.
+ */
+void Load_Forwarded_Functions()
+{
+    static const char dll_name[] = "\\avifil32.dll";
+    char path[MAX_PATH];
+    std::memcpy(path + GetSystemDirectoryA(path, MAX_PATH - sizeof(dll_name)), dll_name, sizeof(dll_name));
+    HMODULE dll = LoadLibraryA(path);
+    if (dll == nullptr) {
+        MessageBoxA(nullptr, "Cannot load original avifil32.dll library", "Proxy", MB_ICONERROR);
+        ExitProcess(0);
+    } else {
+        OriginalAVIStreamWrite = GetProcAddress(dll, "AVIStreamWrite");
+        OriginalAVIFileInit = GetProcAddress(dll, "AVIFileInit");
+        OriginalAVIFileOpenA = GetProcAddress(dll, "AVIFileOpenA");
+        OriginalAVIFileRelease = GetProcAddress(dll, "AVIFileRelease");
+        OriginalAVIFileExit = GetProcAddress(dll, "AVIFileExit");
+        OriginalAVIFileCreateStreamA = GetProcAddress(dll, "AVIFileCreateStreamA");
+        OriginalAVIStreamSetFormat = GetProcAddress(dll, "AVIStreamSetFormat");
+        OriginalAVIStreamRelease = GetProcAddress(dll, "AVIStreamRelease");
+    }
+}
+
+/**
+ * Loads the dll payload and retrieves the Setup_Hooks function from it.
+ */
+bool Load_Setup_Hooks()
+{
+    const char *payload = Get_Payload_Name();
+
+    if (payload == nullptr) {
+        MessageBoxA(nullptr, "Cannot find a payload for this binary", "Proxy", MB_ICONERROR);
```

Comment:
I suspect most users will not understand this message. Maybe can tell something about game.dat?